arroyo/processing/strategies/streaming/transform.py

```diff
@@ -384,17 +425,25 @@ def __check_for_results(self, timeout: Optional[float] = None) -> None:
                 input_batch,
                 self.__pool.apply_async(
                     parallel_transform_worker_apply,
-                    (self.__transform_function, input_batch, output_batch.block, i,),
+                    (
+                        self.__transform_function,
+                        input_batch,
+                        result.output_batch.block,
+                        result.index_processed_til,
+                        bad_messages,
+                    ),
                 ),
             )
             return
 
         logger.debug("Completed %r, reclaiming blocks...", input_batch)
         self.__input_blocks.append(input_batch.block)
-        self.__output_blocks.append(output_batch.block)
+        self.__output_blocks.append(result.output_batch.block)
         self.__batches_in_progress.decrement()
 
         del self.__results[0]
+        if bad_messages:
+            raise InvalidMessages(bad_messages)
```

Comment:
Watch out, this method is called by `join` as well. IS it ok if you raise `InvalidMessages` there ?

Reply:
I think raising it here is fine since `poll` should raise. `join` can be modified to also collect these 'collected' exceptions and then raise one.